src/common/snippets/tests/src/lowered/pass/domain_optimization.cpp

```diff
@@ -0,0 +1,95 @@
+#include "subgraph_simple.hpp"
+#include "lowering_utils.hpp"
+
+namespace ov {
+namespace test {
+namespace snippets {
+DomainOptimizationParams::DomainOptimizationParams(size_t min_jit_work_amount,
+                                                   size_t min_parallel_work_amount,
+                                                   std::vector<ov::PartialShape> input_shapes,
+                                                   ov::snippets::VectorDims exp_master_shape,
+                                                   size_t exp_loop_depth) :
+                                                   min_jit_work_amount(min_jit_work_amount),
+                                                   min_parallel_work_amount(min_parallel_work_amount),
+                                                   input_shapes(std::move(input_shapes)),
+                                                   exp_master_shape(std::move(exp_master_shape)),
+                                                   exp_loop_depth(exp_loop_depth) {
+}
+
+std::string DomainOptimizationTest::getTestCaseName(testing::TestParamInfo<DomainOptimizationParams> obj) {
+    DomainOptimizationParams domain_opt_params = obj.param;
+    std::ostringstream result;
+    result << "MinJitWork=" << domain_opt_params.min_jit_work_amount << "_";
+    result << "MinParWork=" << domain_opt_params.min_parallel_work_amount << "_";
+    for (size_t i = 0; i < domain_opt_params.input_shapes.size(); i++)
+        result << "IS[" << i << "]=" << ov::test::utils::partialShape2str({domain_opt_params.input_shapes[i]}) << "_";
+    result << "ExpMS=" << ov::test::utils::vec2str(domain_opt_params.exp_master_shape) << "_";
+    result << "ExpLD=" << domain_opt_params.exp_loop_depth << "_";
+    return result.str();
+}
+
+void DomainOptimizationTest::SetUp() {
+    m_domain_opt_params = this->GetParam();
+    m_model = std::make_shared<EltwiseFunction>(m_domain_opt_params.input_shapes)->getOriginal();
+}
+
+TEST_P(DomainOptimizationTest, DomainOptimization) {
+    auto subgraph = LoweringTests::getTokenizedSubgraph(m_model);
+    subgraph->set_min_jit_work_amount(m_domain_opt_params.min_jit_work_amount);
+    subgraph->set_min_parallel_work_amount(m_domain_opt_params.min_parallel_work_amount);
+    auto linear_ir = *subgraph->convert_body_to_linear_ir();
+    size_t loop_depth = 1;
+    ov::snippets::lowered::pass::PassPipeline domain_optimization_pipeline;
+    domain_optimization_pipeline.register_pass<ov::snippets::lowered::pass::DomainOptimization>(loop_depth);
+    domain_optimization_pipeline.run(linear_ir);
+    const auto& master_shape = linear_ir.get_master_shape();
+    EXPECT_EQ(loop_depth, m_domain_opt_params.exp_loop_depth) << "Inconsistent loop depth detected";
+    EXPECT_THAT(master_shape, testing::ContainerEq(m_domain_opt_params.exp_master_shape)) << "Inconsistent master_shape detected";
+}
+
+namespace DomainOptimizationTestsInstantiation {
+
+std::vector<DomainOptimizationParams> dopt_params = {
```

Comment:
In general I don't have mentioned concerns.

But I'd suggest to create tests with 1D and 2D shapes. Just in case. What do you think?

Reply:
Good idea! added